### DISCUSSION TITLE
How to use RealSense D455 with OpenVINO to estimate the 3D human pose?

### DISCUSSION THREAD

**Discussion by Hanning-Liu** [openvinotoolkit/openvino_notebooks#2126] [Q&A]
I am using this notebook: https://github.com/openvinotoolkit/openvino_notebooks/blob/latest/notebooks/3D-pose-estimation-webcam/3D-pose-estimation.ipynb
I have connected my D455 camera onto my laptop. But I get the error below:
![image](https://github.com/openvinotoolkit/openvino_notebooks/assets/42676581/de965d21-b258-4d30-894d-bb02af681027)

I think I should add some realsense specific code:
![image](https://github.com/openvinotoolkit/openvino_notebooks/assets/42676581/09fbe6b7-68b5-41ba-8292-0e8cb0116d17)

But I have no idea about what variable should be used as the `source` parameter in the `run_pose_estimation()` method provided by OpenVINO?

**Comment by brmarkus** [comment]
Of course it helps to use Realsense specific APIs "in front of" OpenVINO - like enabling/disabling the laser-dot-projector for some RealSense models (update firmware, getting depth-information, specific formats, etc.).
But in general the underlying layer "video-for-linux-v2" (v4l2) under Linux can be used (selecting the video-mode/resolution/color-format/framerate, etc.).
Are you under Linux...?
Under Linux, you could use `$> v4l2-ctl --list-devices` to find out which devices are currently created: `cam_id` then uses the last digit(s) of `/dev/videoXX`.
Using `$> v4l2-ctl --list-formats-ext -d /dev/video1` for each of the video-devices to see which provides a video-stream in which format, resolution and framerate.

Sorry, I cannot comment for MS-Windows environments...

**Reply by Hanning-Liu** [reply]
I'm using Windows but I have an old NUC with Ubuntu. I will give it a try under Linux system and provide my feedback later. Thanks a lot for your help!

**Reply by brmarkus** [reply]
Searching and found this for MS-Win - but I haven't tested it:
"https://github.com/yushulx/python-capture-device-list"
Not sure it (still) works, last update was 2 years ago.

Other ideas I found are something like this, just iterating over a few devices (here the first 10) and just try if they work and can be read from:

```
def returnCameraIndexes():
    # checks the first 10 indexes.
    index = 0
    arr = []
    i = 10
    while i > 0:
        cap = cv2.VideoCapture(index)
        if cap.read()[0]:
            arr.append(index)
            cap.release()
        index += 1
        i -= 1
    return arr
```
